bxserum/connections/grpc.go

```
@@ -1 +1,43 @@
 package connections
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"io"
+)
+
+func GRPCStream[T any](stream grpc.ClientStream, input string, responseChan chan *T) error {
```

I think you'll want a `ctx` argument for this, so you can cancel the stream by canceling the ctx

---

The reason I did not think I needed to was because the context is already taken when creating a stream. When the context is canceled, RecvMsg should return an error and we will return out of the for loop.

---

gotcha, sounds good :+1: